recirq/quantum_chess/quantum_board.py

```diff
@@ -109,8 +111,13 @@ def __init__(
 
         self.cache = {}
 
-    def with_state(self, basis_state: int) -> "CirqBoard":
-        """Resets the board with a specific classical state."""
+        # Will only be turned on if user specifies
+        self.reset_starting_states = reset_starting_states
+
+    def with_state(self, basis_state: int, reset_move_history=True) -> "CirqBoard":
+        """Resets the board with a specific classical state. reset_move_history indicates
```

Comment:
Probably Could be more clear to put the arguments into its own block:

```
"""Resets the board with a specific classical state.

Args:
    reset_move_history: indicates whether to reset the entire move history of the game. It will be set to false
        if we are calling this function after the board has returned to a fully classical position
"""
```